backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlDatabase.java

```diff
@@ -168,6 +169,18 @@ public void init() { // setup default stuff
       if (getSetting(Constants.PRIVACY_POLICY_TEXT) == null) {
         this.setSetting(Constants.PRIVACY_POLICY_TEXT, Constants.PRIVACY_POLICY_TEXT_DEFAULT);
       }
+
+      String key =
+          (String)
+              EnvironmentProperty.getParameter(Constants.MOLGENIS_JWT_SHARED_SECRET, null, STRING);
+      if (key == null) {
+        key = getSetting(Constants.MOLGENIS_JWT_SHARED_SECRET);
```

Comment:
getSetting(Constants.MOLGENIS_JWT_SHARED_SECRET) always returns null after server restart, even if there is a key present. This results in a new key being generated and thereby invalidating existing tokens.